x-pack/plugins/fleet/server/services/agent_policy.ts

```diff
@@ -715,31 +743,66 @@ class AgentPolicyService {
       return;
     }
 
-    const policy = await agentPolicyService.get(soClient, agentPolicyId);
-    const fullPolicy = await agentPolicyService.getFullAgentPolicy(soClient, agentPolicyId);
-    if (!policy || !fullPolicy || !fullPolicy.revision) {
-      return;
-    }
+    const policies = await agentPolicyService.getByIDsAsMap(soClient, agentPolicyIds);
+    const fullPolicies = await Promise.all(
+      agentPolicyIds.map((agentPolicyId) =>
+        // TODO: make this a bulk
```

Comment:
```suggestion
        // There are some potential performance concerns around using `getFullAgentPolicy` in this context, e.g.
        // re-fetching outputs, settings, and upgrade download source URI data for each policy. This could potentially
        // be a bottleneck in environments with several thousand agent policies being deployed here.
```